src/auth/auth.service.ts

```diff
@@ -0,0 +1,377 @@
+      }),
+    );
+
+    await this.mailService.sendTemplateMail(
+      {
+        to: email,
+        subject: 'Welcome to Audease',
+      },
+      'school-onboarding',
+      {
+        first_name,
+        last_name,
+        college_name,
+        onboardingKey,
+      },
+    );
+
+    return {
+      message:
+        'School created successfully check your mail for further instructions',
+      keyId: onboardingKey,
+    };
+  }
+
+  async verifySchool(key: string) {
+    const data = await this.redis.hget('onboarding', key);
+
+    if (!data) {
+      this.logger.error('Invalid key');
+      throw new NotFoundException('Invalid key');
+    }
+
+    const { email, first_name, last_name, college_id } = JSON.parse(data);
+
+    const school = await this.authRepository.updateStatus(
+      college_id,
+      RegistrationStatus.VERIFIED,
+    );
+
+    const verifyUrl = `${process.env.FRONTEND_URL}/verify?token=${key}`;
+
+    const school_name = school.college_name;
+
+    await this.mailService.sendTemplateMail(
+      {
+        to: email,
+        subject: 'School Verification Successful',
+      },
+      'sucessful-verification',
+      {
+        first_name,
+        last_name,
+        school_name,
+        verifyUrl,
+      },
+    );
+    return {
+      message: 'School verified successfully',
+    };
+  }
+
+  async verifyKey(key: string) {
+    const data = await this.redis.hget('onboarding', key);
+
+    if (!data) {
+      this.logger.error('Invalid key');
+      throw new NotFoundException('Invalid key');
+    }
+
+    return {
+      message: 'Key verified successfully',
+    };
+  }
+
+  async createUser(data: IUserCreate) {
+    const { username, password, keyId } = data;
+
+    const onboardingData = await this.redis.hget('onboarding', keyId);
+
+    if (!onboardingData) {
+      this.logger.error('Invalid key');
+      throw new NotFoundException('Invalid key');
+    }
+
+    const { email, first_name, last_name, college_id, phone } =
+      JSON.parse(onboardingData);
+
+    const role = await this.userService.getRoleByName(Role.SCHOOL_ADMIN);
+
+    const userExists = await this.userService.getUserByUsername(username);
+
+    if (userExists) {
+      this.logger.error('Username already exists');
+      throw new ConflictException('Username already exists');
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const user = await this.userService.createUserWithCollegeId(
+      {
+        username,
+        password: await bcrypt.hash(password, 10),
+        email,
+        phone,
+        first_name,
+        last_name,
+        role,
+      },
+      college_id,
+    );
+
+    // Remove the onboarding key from redis
+    await this.redis.hdel('onboarding', keyId);
+
+    return {
+      message: 'User created successfully',
+    };
+  }
+
+  async addRecruiter(userId: string, createUserDto: CreateAccountDto) {
+    const admin = await this.userRepository.findOne({
```

Comment:
There is already a auth.repository.ts for database methods. I thought we were already clear on using Repository Pattern na?